test/unit/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuppressCaseControllerSpec.scala

```diff
@@ -106,52 +108,59 @@ class SuppressCaseControllerSpec extends WordSpec with Matchers with UnitSpec
-      when(casesService.suppressCase(refEq(caseWithStatusNEW), refEq(operator))(any[HeaderCarrier])).thenReturn(successful(caseWithStatusSUPRRESSED))
+    def aMultipartFileWithParams(params: (String, Seq[String])*): MultipartFormData[TemporaryFile] = {
+      val file = TemporaryFile("example-file.txt")
+      val filePart = FilePart[TemporaryFile](key = "email", "file.txt", contentType = Some("text/plain"), ref = file)
+      MultipartFormData[TemporaryFile](dataParts = params.toMap, files = Seq(filePart), badParts = Seq.empty)
+    }
+
+    def anEmptyMultipartFileWithParams(params: (String, Seq[String])*): MultipartFormData[TemporaryFile] = {
+      val filePart = FilePart[TemporaryFile](key = "email", "", contentType = Some("text/plain"), ref = TemporaryFile("example-file.txt"))
+      MultipartFormData[TemporaryFile](dataParts = params.toMap, files = Seq(filePart), badParts = Seq.empty)
+    }
+
+    "redirect to confirmation" in {
+      when(casesService.suppressCase(refEq(caseWithStatusNEW), any[FileUpload], refEq("some-note"), refEq(operator))(any[HeaderCarrier])).thenReturn(successful(caseWithStatusSUPRRESSED))
 
       val result: Result =
         await(controller(caseWithStatusNEW).postSuppressCase("reference")
-             (newFakePOSTRequestWithCSRF(fakeApplication).withFormUrlEncodedBody("state" -> "true")))
+             (newFakePOSTRequestWithCSRF(fakeApplication).withBody(aMultipartFileWithParams("note" -> Seq("some-note")))))
 
       status(result) shouldBe Status.SEE_OTHER
       locationOf(result) shouldBe Some("/tariff-classification/cases/reference/suppress/confirmation")
     }
 
-    "redirect to contact info when operator indicates that applicant has not been contacted" in {
-      when(casesService.suppressCase(refEq(caseWithStatusNEW), refEq(operator))(any[HeaderCarrier])).thenReturn(successful(caseWithStatusSUPRRESSED))
+    "return to form on missing file" in {
+      val result: Result =
+        await(controller(caseWithStatusNEW).postSuppressCase("reference")
+        (newFakePOSTRequestWithCSRF(fakeApplication).withBody(anEmptyMultipartFileWithParams())))
 
+      status(result) shouldBe Status.OK
+      bodyOf(result) should include("Change the status of this case to: Suppressed")
+    }
+
+    "return to form on missing form field" in {
       val result: Result =
         await(controller(caseWithStatusNEW).postSuppressCase("reference")
-             (newFakePOSTRequestWithCSRF(fakeApplication).withFormUrlEncodedBody("state" -> "false")))
+        (newFakePOSTRequestWithCSRF(fakeApplication).withBody(aMultipartFileWithParams())))
 
-      status(result) shouldBe Status.SEE_OTHER
-      locationOf(result) shouldBe Some("/tariff-classification/cases/reference/suppress/contact-info")
+      status(result) shouldBe Status.OK
+      bodyOf(result) should include("Change the status of this case to: Suppressed")
     }
 
```

Comment:
Should we also be testing here for return to form on file exceeded max size, and file was unsupported type?

Reply:
I pulled the stuff you did across